File: src/test/java/ml/wonwoo/testcontainerstest/mongo/MongoDbContainerInitializer.java
package ml.wonwoo.testcontainerstest.mongo;

import org.springframework.boot.test.util.TestPropertyValues;
import org.springframework.context.ApplicationContextInitializer;
import org.springframework.context.ConfigurableApplicationContext;

public class MongoDbContainerInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {

    private JMongoDbContainer mongoDbContainer = new JMongoDbContainer();

    @Override
    public void initialize(ConfigurableApplicationContext applicationContext) {
        mongoDbContainer.start();

        TestPropertyValues.of(

                "spring.data.mongodb.uri=mongodb://" + mongoDbContainer.getContainerIpAddress() + ":" + mongoDbContainer.getPort() + "/test"

        ).applyTo(applicationContext);
    }
}

File: src/test/kotlin/ml/wonwoo/testcontainerstest/mongo/MongoDbContainerInitializer.kt
package ml.wonwoo.testcontainerstest.mongo

import ml.wonwoo.testcontainerstest.MongoDbContainer
import org.springframework.boot.test.util.TestPropertyValues
import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.ConfigurableApplicationContext

class MongoDbContainerInitializer : ApplicationContextInitializer<ConfigurableApplicationContext> {

    private val mongoContainer: MongoDbContainer = MongoDbContainer()

    override fun initialize(applicationContext: ConfigurableApplicationContext) {

        mongoContainer.start()

        TestPropertyValues.of(

            "spring.data.mongodb.uri=mongodb://${mongoContainer.containerIpAddress}:${mongoContainer.getPort()}/test"

        ).applyTo(applicationContext)
    }
}